### ISSUE TITLE
Rolling restart of the monitor services gets stuck with multiple monitor nodes

### ISSUE THREAD

**Issue by shinnosuke-okada** [pingcap/tidb-operator#668]
## Bug Report

**What version of Kubernetes are you using?**
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-20T04:49:16Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.7-eks-c57ff8", GitCommit:"c57ff8e35590932c652433fab07988da79265d5b", GitTreeState:"clean", BuildDate:"2019-06-07T20:43:03Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

**What version of TiDB Operator are you using?**
```
$ kubectl exec -n tidb-admin tidb-controller-manager-7b998f5747-wltsw -- tidb-controller-manager -V
TiDB Operator Version: version.Info{TiDBVersion:"2.1.0", GitVersion:"v1.0.0-beta.3", GitCommit:"6257dfaad68f55f745f20f6f5d19b10bea2b0bea", GitTreeState:"clean", BuildDate:"2019-06-06T09:51:04Z", GoVersion:"go1.12", Compiler:"gc", Platform:"linux/amd64"}
```

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
```
$ kubectl get sc
NAME            PROVISIONER                    AGE
ebs-gp2         kubernetes.io/aws-ebs          5h14m
gp2 (default)   kubernetes.io/aws-ebs          5h38m
local-storage   kubernetes.io/no-provisioner   5h14m
$ kubectl get pvc -n stg5tidb
NAME                      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS    AGE
data-stg5tidb-drainer-0   Bound    pvc-6f4cfb1d-a940-11e9-82ce-06a870c66dd2   300Gi      RWO            ebs-gp2         167m
data-stg5tidb-pump-0      Bound    pvc-6f5358a9-a940-11e9-82ce-06a870c66dd2   400Gi      RWO            ebs-gp2         167m
data-stg5tidb-pump-1      Bound    pvc-7f3a1ce7-a940-11e9-82ce-06a870c66dd2   400Gi      RWO            ebs-gp2         167m
data-stg5tidb-pump-2      Bound    pvc-94300704-a940-11e9-82ce-06a870c66dd2   400Gi      RWO            ebs-gp2         166m
pd-stg5tidb-pd-0          Bound    pvc-ccda72c5-a930-11e9-82ce-06a870c66dd2   1Gi        RWO            ebs-gp2         4h39m
pd-stg5tidb-pd-1          Bound    pvc-ccdcce54-a930-11e9-82ce-06a870c66dd2   1Gi        RWO            ebs-gp2         4h39m
pd-stg5tidb-pd-2          Bound    pvc-cce0e49b-a930-11e9-82ce-06a870c66dd2   1Gi        RWO            ebs-gp2         4h39m
stg5tidb-monitor          Bound    pvc-ccad322a-a930-11e9-91f3-0a75f9078dde   100Gi      RWO            ebs-gp2         4h39m
tikv-stg5tidb-tikv-0      Bound    local-pv-3e2bc74d                          1741Gi     RWO            local-storage   4h38m
tikv-stg5tidb-tikv-1      Bound    local-pv-440682c6                          1741Gi     RWO            local-storage   4h38m
tikv-stg5tidb-tikv-2      Bound    local-pv-d46097e6                          1741Gi     RWO            local-storage   4h38m
```

**What's the status of the TiDB cluster pods?**
```
$ kubectl get po -n stg5tidb -o wide
NAME                                  READY   STATUS     RESTARTS   AGE     IP               NODE                                                NOMINATED NODE   READINESS GATES
stg5tidb-discovery-7bbc4458f8-d8s5p   1/1     Running    0          4h37m   10.223.85.126    ip-10-223-68-4.ap-northeast-1.compute.internal      <none>           <none>
stg5tidb-drainer-0                    1/1     Running    1          165m    10.223.87.183    ip-10-223-68-4.ap-northeast-1.compute.internal      <none>           <none>
stg5tidb-monitor-5484c599cb-s52vn     2/2     Running    0          155m    10.223.49.118    ip-10-223-47-120.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-monitor-5f88c8855b-26m8q     0/2     Init:0/1   0          38m     <none>           ip-10-223-47-117.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pd-0                         1/1     Running    0          4h37m   10.223.115.158   ip-10-223-89-237.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pd-1                         1/1     Running    0          4h37m   10.223.187.149   ip-10-223-149-28.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pd-2                         1/1     Running    1          4h37m   10.223.2.223     ip-10-223-42-81.ap-northeast-1.compute.internal     <none>           <none>
stg5tidb-pump-0                       1/1     Running    0          165m    10.223.143.32    ip-10-223-142-52.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pump-1                       1/1     Running    0          165m    10.223.11.47     ip-10-223-47-120.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pump-2                       1/1     Running    0          164m    10.223.104.150   ip-10-223-68-4.ap-northeast-1.compute.internal      <none>           <none>
stg5tidb-tidb-0                       1/1     Running    0          164m    10.223.124.55    ip-10-223-74-44.ap-northeast-1.compute.internal     <none>           <none>
stg5tidb-tidb-1                       1/1     Running    3          165m    10.223.56.247    ip-10-223-16-49.ap-northeast-1.compute.internal     <none>           <none>
stg5tidb-tikv-0                       1/1     Running    0          4h36m   10.223.26.86     ip-10-223-34-104.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-tikv-1                       1/1     Running    0          4h36m   10.223.181.151   ip-10-223-159-213.ap-northeast-1.compute.internal   <none>           <none>
stg5tidb-tikv-2                       1/1     Running    0          4h36m   10.223.85.9      ip-10-223-107-229.ap-northeast-1.compute.internal   <none>           <none>
```

**What did you do?**
1. Deploy TiDB operator and TiDB cluster on AWS
2. Provision multiple nodes for the `monitor` service in ASG
3. Enable persistence for the `monitor` service
4. Rolling restart the `monitor` service

**What did you expect to see?**
Restart successfully without getting stuck.

**What did you see instead?**
The new ReplicaSet that gets created is stuck due to `ReadWriteOnce` EBS.
```
$ kubectl describe pod stg5tidb-monitor-5f88c8855b-26m8q -n stg5tidb
Name:           stg5tidb-monitor-5f88c8855b-26m8q
Namespace:      stg5tidb
Priority:       0
Node:           ip-10-223-47-117.ap-northeast-1.compute.internal/10.223.47.117
Start Time:     Thu, 18 Jul 2019 20:49:49 +0900
Labels:         app.kubernetes.io/component=monitor
                app.kubernetes.io/instance=stg5tidb
                app.kubernetes.io/name=tidb-cluster
                pod-template-hash=5f88c8855b
Annotations:    kubectl.kubernetes.io/restartedAt: 2019-07-18T20:49:49+09:00
                kubernetes.io/psp: eks.privileged
Status:         Pending
IP:
Controlled By:  ReplicaSet/stg5tidb-monitor-5f88c8855b
Init Containers:
  init-data:
    Container ID:
    Image:         prom/prometheus:v2.2.1
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c
      mkdir -p /data/prometheus /data/grafana
      chmod 777 /data/prometheus /data/grafana
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /data from monitor-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from stg5tidb-monitor-token-mxdk7 (ro)
Containers:
  prometheus:
    Container ID:
    Image:         prom/prometheus:v2.2.1
    Image ID:
    Port:          9090/TCP
    Host Port:     0/TCP
    Command:
      /bin/prometheus
      --web.enable-admin-api
      --log.level=info
      --config.file=/etc/prometheus/prometheus.yml
      --storage.tsdb.path=/data/prometheus
      --storage.tsdb.retention=12d
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:
      TZ:  UTC
    Mounts:
      /data from monitor-data (rw)
      /etc/prometheus from prometheus-config (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from stg5tidb-monitor-token-mxdk7 (ro)
  grafana:
    Container ID:
    Image:          grafana/grafana:6.0.1
    Image ID:
    Port:           3000/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:
      GF_PATHS_DATA:               /data/grafana
      GF_SECURITY_ADMIN_USER:      <set to the key ‘username’ in secret ‘stg5tidb-monitor’>  Optional: false
      GF_SECURITY_ADMIN_PASSWORD:  <set to the key ‘password’ in secret ‘stg5tidb-monitor’>  Optional: false
      GF_AUTH_ANONYMOUS_ENABLED:   true
      GF_AUTH_ANONYMOUS_ORG_NAME:  Main Org.
      GF_AUTH_ANONYMOUS_ORG_ROLE:  Viewer
      TZ:                          UTC
    Mounts:
      /data from monitor-data (rw)
      /etc/grafana/provisioning/dashboards from dashboards-provisioning (rw)
      /etc/grafana/provisioning/datasources from datasource (rw)
      /grafana-dashboard-definitions/tidb from grafana-dashboard (rw)
      /tmp from grafana-prestart (rw)
      /tmp/dashboard-v2 from dashboard-v2 (rw)
      /tmp/dashboard-v3 from dashboard-v3 (rw)
      /tmp/dashboard-v3/extra from dashboard-extra-v3 (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from stg5tidb-monitor-token-mxdk7 (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  monitor-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  stg5tidb-monitor
    ReadOnly:   false
  prometheus-config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      stg5tidb-monitor
    Optional:  false
  datasource:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      stg5tidb-monitor
    Optional:  false
  dashboards-provisioning:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      stg5tidb-monitor
    Optional:  false
  grafana-dashboard:
    Type:       EmptyDir (a temporary directory that shares a pod’s lifetime)
    Medium:
    SizeLimit:  <unset>
  dashboard-v2:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      stg5tidb-monitor-dashboard-v2
    Optional:  false
  dashboard-v3:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      stg5tidb-monitor-dashboard-v3
    Optional:  false
  dashboard-extra-v3:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      stg5tidb-monitor-dashboard-extra-v3
    Optional:  false
  grafana-prestart:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      stg5tidb-monitor
    Optional:  false
  stg5tidb-monitor-token-mxdk7:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  stg5tidb-monitor-token-mxdk7
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason              Age   From                     Message
  ----     ------              ----  ----                     -------
  Normal   Scheduled           22s   default-scheduler        Successfully assigned stg5tidb/stg5tidb-monitor-5f88c8855b-26m8q to ip-10-223-47-117.ap-northeast-1.compute.internal
  Warning  FailedAttachVolume  22s   attachdetach-controller  Multi-Attach error for volume “pvc-ccad322a-a930-11e9-91f3-0a75f9078dde” Volume is already used by pod(s) stg5tidb-monitor-5484c599cb-s52vn
```
```
$ kubectl get pod -o wide -n stg5tidb
NAME                                  READY   STATUS     RESTARTS   AGE     IP               NODE                                                NOMINATED NODE   READINESS GATES
stg5tidb-discovery-7bbc4458f8-d8s5p   1/1     Running    0          4h7m    10.223.85.126    ip-10-223-68-4.ap-northeast-1.compute.internal      <none>           <none>
stg5tidb-drainer-0                    1/1     Running    1          135m    10.223.87.183    ip-10-223-68-4.ap-northeast-1.compute.internal      <none>           <none>
stg5tidb-monitor-5484c599cb-s52vn     2/2     Running    0          125m    10.223.49.118    ip-10-223-47-120.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-monitor-5f88c8855b-26m8q     0/2     Init:0/1   0          8m36s   <none>           ip-10-223-47-117.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pd-0                         1/1     Running    0          4h7m    10.223.115.158   ip-10-223-89-237.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pd-1                         1/1     Running    0          4h7m    10.223.187.149   ip-10-223-149-28.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pd-2                         1/1     Running    1          4h7m    10.223.2.223     ip-10-223-42-81.ap-northeast-1.compute.internal     <none>           <none>
stg5tidb-pump-0                       1/1     Running    0          135m    10.223.143.32    ip-10-223-142-52.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pump-1                       1/1     Running    0          134m    10.223.11.47     ip-10-223-47-120.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-pump-2                       1/1     Running    0          134m    10.223.104.150   ip-10-223-68-4.ap-northeast-1.compute.internal      <none>           <none>
stg5tidb-tidb-0                       1/1     Running    0          134m    10.223.124.55    ip-10-223-74-44.ap-northeast-1.compute.internal     <none>           <none>
stg5tidb-tidb-1                       1/1     Running    3          135m    10.223.56.247    ip-10-223-16-49.ap-northeast-1.compute.internal     <none>           <none>
stg5tidb-tikv-0                       1/1     Running    0          4h6m    10.223.26.86     ip-10-223-34-104.ap-northeast-1.compute.internal    <none>           <none>
stg5tidb-tikv-1                       1/1     Running    0          4h6m    10.223.181.151   ip-10-223-159-213.ap-northeast-1.compute.internal   <none>           <none>
stg5tidb-tikv-2                       1/1     Running    0          4h6m    10.223.85.9      ip-10-223-107-229.ap-northeast-1.compute.internal   <none>           <none>
```

**Comment by weekface** [comment]
Will change monitor deployment strategy to `Recreate`: https://github.com/pingcap/tidb-operator/pull/666/files#diff-3b3eccb8f3c8f4c945f46efc1ace69c1R14-R15

**Comment by aylei** [comment]
closed via #666
